tests/test_config.py

```diff
@@ -233,6 +231,7 @@ def test_config_modified_settings(self, tmp_path_home):
     max_ctx_size: 2048
     llm_family: ''
   vllm:
+    gpus: 8
```

Comment:
This is ok, but the test is a bit limited, since it only checks that `vllm_args` on config object are set. But then nothing checks that they are propagated down to vllm backend. I think it may be a problem (see my other comment in `serve.py`.)

Reply:
Resolving based on new tests